### ISSUE TITLE
Ability to show gaps in data?

### ISSUE THREAD

**Issue by stx** [imaNNeoFighT/fl_chart#26]
Again, thanks for this plugin. You're doing an awesome job!

Is it possible to show gaps or null data? I've played around and haven't found how one might do that.

Visually, something along these lines:

![image](https://user-images.githubusercontent.com/1284134/59556416-0c321600-8f77-11e9-938e-1c941a2e5937.png)

Implementation wise, this might look like:

```
                    FlSpot(1, 2),
                    FlSpot(2, 3),
                    FlSpot(3, 3),
                    FlSpot(4, null),
                    FlSpot(5, null),
                    FlSpot(6, 2),
                    FlSpot(7, 4),
```

Thanks again!

**Comment by imaNNeoFighT** [comment]
You can have multiple bars on the LineChart, then make two separted and combine them in your chart,

**Comment by imaNNeoFighT** [comment]
check [LineBarsData] (https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/line_chart.md#linechartdata), you can have multiple [LineChartBarData] (https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/line_chart.md#linechartbardata)

**Comment by stx** [comment]
Got it! Thank you!

**Comment by shamilovtim** [comment]
Hello @imaNNeoFighT,

Thanks for the library! It's great so far. Is it possible to get real nullable coordinates added without having to create multiple instances of lines? This is a standard feature of a lot of different charting libraries.

In almost all of them the best practice is:
```json
{ x: 5, y: null },
{ x: 6, y: null }
```

See:
1. https://google.github.io/charts/flutter/example/line_charts/simple_nulls
2. https://google.github.io/charts/flutter/example/line_charts/simple_nulls

**Comment by imaNNeoFighT** [comment]
Okay fine, we need this in the line chart, but It is weird when we have rounded line chart

**Comment by shamilovtim** [comment]
Definitely but I think it conveys the proper feeling from the business logic side. The users didn't chart something that they were supposed to do, which caused the problem.

We could offer settings to interpolate with a dashed line and/or a pure null option:

Pure null
<img width="368" alt="Screen Shot 2020-01-10 at 2 29 42 PM" src="https://user-images.githubusercontent.com/3495974/72180690-ac3d0a00-33b5-11ea-8365-776cac1e6b1a.png">

Interpolated with a dashed line:
<img width="348" alt="Screen Shot 2020-01-10 at 2 30 14 PM" src="https://user-images.githubusercontent.com/3495974/72180725-bd861680-33b5-11ea-96a3-853bbd70b598.png">

I would be happy to help with this feature if we go with flutter for our project

**Comment by imaNNeoFighT** [comment]
Nice, we will consider it.
Thanks for reporting!

**Comment by yongjhih** [comment]
That's what I did for nullable spots:

```dart
    final baseLineChartBarData = LineChartBarData(
      isCurved: true,
      preventCurveOverShooting: true,
      // ...
    );

    final Iterable<FlSpot> spotsNullable = values
        .mapIndexed((i, value) => FlSpot(i.toDouble(), value));

    final List<List<FlSpot>> spotsList = spotsNullable
        .fold<List<List<FlSpot>>>(<List<FlSpot>>[<FlSpot>[]], (that, it) {
          if (it.y != null) {
            that.last.add(it);
          } else {
            if (that.last.isNotEmpty) {
              that.add([]);
            }
          }
          return that;
        });
    final List<LineChartBarData> lineBarsData =
      spotsList.where((it) => it.isNotEmpty).map((List<FlSpot> _spots) => baseLineChartBarData.copyWith(spots: _spots)).toList();
    return LineChart(LineChartData(
        lineBarsData: lineBarsData,
        // ...
    ));
```

**Comment by shamilovtim** [comment]
@yongjhih nulls should be in the next release

**Comment by imaNNeoFighT** [comment]
Implemented in [0.8.6](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.8.6), check it out!

**Comment by cryosx** [comment]
> How can I use null values? If null is parsed as Y value, the following exception is thrown:
> 
> `Unhandled Exception: type 'Null' is not a subtype of type 'double'`
> 
> My use case requires valid X values with null Y values

FlSpot.nullSpot)
